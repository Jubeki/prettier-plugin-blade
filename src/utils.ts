import { format } from "prettier";
// @ts-ignore
import php from "@prettier/plugin-php/standalone";

export const formatAsPhp = (source: string): string => {
  if (!source.startsWith("<?php")) {
    source = "<?php " + source;
  }

  let code = format(source, { parser: "php", plugins: [php] })
    .replace("<?php ", "")
    .trim();

  if (source.trim().endsWith(";")) {
    return code;
  }

  // The PHP plugin for Prettier will add a semi-colon by default. We don't always want that.
  return code.substring(0, code.length - 1);
};
