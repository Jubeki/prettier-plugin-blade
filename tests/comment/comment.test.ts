import { formatFile } from "../utils";
import path from "path";

test("basic", () => {
  const formatted = formatFile(path.join(__dirname, "./comment.blade.php"));
  expect(formatted).toMatchSnapshot();
});

it("does not escape comments", () => {
  const formatted = formatFile(
    path.join(__dirname, "./comment-with-escape.blade.php")
  );
  expect(formatted).toMatchSnapshot();
});
