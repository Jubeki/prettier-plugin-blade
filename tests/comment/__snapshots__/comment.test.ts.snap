// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"<h1>Here's some very nice text {{-- This is actually not that nice --}}</h1>
"
`;

exports[`does not escape comments 1`] = `
"<h1>Here's some very nice text @{{--This is actually not that nice--}}</h1>
"
`;
